tui/tui.go

```diff
@@ -105,13 +106,30 @@ func (ui *UI) Subscribe(events <-chan taskrunner.ExecutorEvent) error {
 			ui.Update(func() {
 				tab.view.Append(ev.Message)
 			})
+		case *taskrunner.TaskInvalidatedEvent:
+			ui.Update(func() {
+				tab.view.Append(fmt.Sprintf("Invalidating for %d reasons:", len(ev.Reasons)))
+				for _, reason := range ev.Reasons {
+					tab.view.Append(fmt.Sprintf("- %s", reason.Description()))
+				}
+			})
 		case *taskrunner.TaskCompletedEvent:
 			ui.Update(func() {
-				tab.view.Append(fmt.Sprintf("Completed! (%d)", ev.Duration))
+				var msg string
+				if ev.Duration == 0 {
+					msg = "Completed"
+				} else {
+					msg = fmt.Sprintf("Completed (%0.2fs)", float64(ev.Duration)/float64(time.Second))
+				}
+				tab.view.Append(msg)
 			})
 		case *taskrunner.TaskFailedEvent:
 			ui.Update(func() {
-				tab.view.Append(fmt.Sprintf("Failed! (%v)", ev.Error))
+				tab.view.Append(fmt.Sprintf("Failed\n%v", ev.Error))
```

Comment:
should we append twice vs. `\n`? Not sure if the api does something different.

Reply:
We can do that. Doesn't make a huge difference where the tab view is concerned.